modules/wishlist-magento/repositories/useAddToWishlistRepository.ts

```diff
@@ -1,20 +1,35 @@
 import useProduct from '#ioc/composables/useProduct'
 import useMagento from '#ioc/composables/useMagento'
 import AddProductsToWishlist from '#ioc/graphql/mutations/AddProductsToWishlist'
+import ToWishlistUserError from '#ioc/mappers/ToWishlistUserError'
 import useWishlistStore from '#ioc/stores/useWishlistStore'
 
 export default () => {
   const magento = useMagento()
   const wishlistStore = useWishlistStore()
 
-  return async (product: ReturnType<typeof useProduct>) => {
+  return async (
+    product: ReturnType<typeof useProduct>,
+  ): Promise<{
```

Comment:
Dont change the type of the repository. Instead handle the userErrors field as regular errors and throw them.